components/src/banner.ts

```diff
@@ -0,0 +1,292 @@
+import {
+  html,
+  unsafeCSS,
+  LitElement,
+  type ReactiveController,
+  type ReactiveControllerHost
+} from 'lit'
+import { property, state } from 'lit/decorators.js'
+import defaultLogo from './assets/wm_logo_animated.svg?url'
+import bannerStyles from './banner.css?raw'
+
+export interface BannerConfig {
+  bannerTitleText?: string
+  bannerDescriptionText?: string
+  bannerBorderRadius?: string
+  bannerPosition?: 'Top' | 'Bottom'
+  bannerSlideAnimation?: 'Down' | 'None'
+  theme?: {
+    primaryColor?: string
+    backgroundColor?: string
+    textColor?: string
+    fontFamily?: string
+    fontSize?: number
+  }
+  logo?: string
+}
+
+export class Banner extends LitElement {
+  private configController = new BannerController(this)
+
+  @property({ type: Object })
+  set config(value: Partial<BannerConfig>) {
+    this.configController.updateConfig(value)
+  }
+  get config() {
+    return this.configController.config
+  }
+  @property({ type: Boolean }) isVisible = true
+
+  @state() private isDismissed = false
+  @state() private isAnimating = false
+  @state() private animationClass = ''
+  connectedCallback() {
+    super.connectedCallback()
+    this.previewAnimation()
+  }
+
+  static styles = unsafeCSS(bannerStyles)
+  private handleClose() {
+    this.isDismissed = true
+    this.requestUpdate()
+
+    this.dispatchEvent(
+      new CustomEvent('banner-closed', {
+        detail: { dismissed: true },
+        bubbles: true,
+        composed: true
+      })
+    )
+  }
+
+  private handleLinkClick() {
+    const getWebMonetizationLinkHref = () => {
+      const userAgent = navigator.userAgent
+      if (userAgent.includes('Firefox')) {
+        return 'https://addons.mozilla.org/en-US/firefox/addon/web-monetization-extension/'
+      } else if (
+        userAgent.includes('Chrome') &&
+        !userAgent.includes('Edg') &&
+        !userAgent.includes('OPR')
+      ) {
+        return 'https://chromewebstore.google.com/detail/web-monetization/oiabcfomehhigdepbbclppomkhlknpii'
+      } else if (userAgent.includes('Edg')) {
+        return 'https://microsoftedge.microsoft.com/addons/detail/web-monetization/imjgemgmeoioefpmfefmffbboogighjl'
+      }
+      return 'https://webmonetization.org/'
+    }

```

Comment:
Create a util.ts and `export const getWebMonetizationLinkHref = (userAgent: string) => {}` from there.